File: ProgrammingBasicsJava/Jan2020/LoopsPart2Exercise/src/_02ExamPreparation.java
import java.util.Scanner;

public class _02ExamPreparation {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        int badNotes = Integer.parseInt(scanner.nextLine());
        String input = scanner.nextLine();
        int countTasks = 0;
        double sumNotes = 0;
        int totalBadNotes = 0;
        String task = "";

        while (!input.equals("Enough") && totalBadNotes < badNotes) {
            task = input;
            double note = Double.parseDouble(scanner.nextLine());
            sumNotes += note;
            countTasks++;

            if (note <= 4.00) {
                totalBadNotes++;

                if (totalBadNotes == badNotes) break;
            }

            input = scanner.nextLine();
        }

        double averageNote = sumNotes / countTasks;

        if (totalBadNotes < badNotes) {
            System.out.printf("Average score: %.2f%n" +
                            "Number of problems: %d%n" +
                            "Last problem: %s%n",
                    averageNote,
                    countTasks,
                    task);
        }
        else {
            System.out.printf("You need a break, %d poor grades.", totalBadNotes);
        }
    }
}

File: JavaScriptBasics/05_2_WhileLoopExercise/02_ExamPreparation.js
function printExamScore(input) {
    let index = 0;
    let allowedBadScores = Number(input[index]);
    let countBadScores = 0;
    let sumScores = 0;
    let averageScore = 0;
    let countOfTasks = 0;
    let lastTask = "";


    
    while (allowedBadScores > 0) {
        let taskName = input[index + 1];

        if (taskName === "Enough") {
            console.log(`Average score: ${averageScore}`);
            console.log(`Number of problems: ${countOfTasks}`);
            console.log(`Last problem: ${lastTask}`);
            return;
        }

        let score = Number(input[index + 2]);

        if (score <= 4) {
            countBadScores++;
            allowedBadScores--;
        }

        sumScores += score;
        countOfTasks++;
        averageScore = (sumScores / countOfTasks).toFixed(2);
        lastTask = taskName;

        index += 2;
    }

    console.log(`You need a break, ${countBadScores} poor grades.`);
}

printExamScore(["2",
"Income",
"3",
"Game Info",
"6",
"Best Player",
"4"]);